### PR TITLE
CORTX-31921 : purge data feature branch test updated

### PR DESCRIPTION
**Pull request by RajeshPratapSInghChouhan**
# Problem Statement
[CORTX-31921](https://jts.seagate.com/browse/CORTX-31921) : [F-71A] TE : purge data feature branch testing

# Design
[CORTX-30001](https://jts.seagate.com/browse/CORTX-30001)

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [x] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [x] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [x] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by RajeshPratapSInghChouhan** [comment]
pytest --local True --collect-only
tests/security/test_cortx_port_scanner_netstat.py::test_cortx_port_scanner_netstat', 'test_id': 'TEST-34218', 'marks': ['security', 'skip']}] created at /root/jenkins/workspace/QA/QA_R2_Single_Runner_CSM_error_msg_check/cortx-test/log/te_meta.json
Successfully unmounted directory

====================================================================== 2252 tests collected in 15.77s =======================================================================
# git branch
* CORTX-31921_F-71A-TE_purge_data_feature_branch_testing
  main

**Comment by RajeshPratapSInghChouhan** [comment]
TE & Jobs : 
[TEST-35138](https://jts.seagate.com/browse/TEST-35138)
[Job](https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/72)
[TEST-29872](https://jts.seagate.com/browse/TEST-29872)
[Job](https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/73)
[TEST-28687](https://jts.seagate.com/browse/TEST-28687)
[Job](https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/74)
[TEST-40758](https://jts.seagate.com/browse/TEST-40758)
[Job](https://eos-jenkins.colo.seagate.com/view/QA-R2/job/QA/job/QA_R2_Single_Runner_CSM_error_msg_check/75)

**Review by kachhwahadivya — APPROVED** [review]

**Review by apoorva-rao — APPROVED** [review]
@RajeshPratapSInghChouhan please verify CI/CD sanity and regression runs are all successful

File: libs/csm/rest/csm_rest_iamuser.py [hunk 1]
```diff
@@ -558,7 +558,7 @@ def remove_user_caps_rgw(self, uid, payload: dict):
         return response
 
     @RestTestLib.authenticate_and_login
-    def delete_iam_user_rgw(self, uid, header, purge_data=False):
+    def delete_iam_user_rgw(self, uid, header, purge_data=None):
```
Comment (seagate-sarang-sawant) [inline libs/csm/rest/csm_rest_iamuser.py:561]: purge_data=False was fine.

**Review by seagate-sarang-sawant — APPROVED** [review]

File: libs/csm/rest/csm_rest_iamuser.py (hunk 1 above)
Comment (RajeshPratapSInghChouhan) [reply libs/csm/rest/csm_rest_iamuser.py:561]: all options are getting ignored with delete user

> [Yesterday 1:22 PM] Rajesh Pratap Singh Chouhan
Rohit are we ignoring all option & "purge_data" in delete REST api with this changes ?i.e. earlier we use to get error incase we gave {"purge_data": abc} or {"abc": abc} with the API..but now we are not considering anything at all so not getting any error for this ? or it's a bug ??cc Pranay Pawan

> [Yesterday 1:31 PM] Pranay Kumar
I think no as per standard ... All extra inputs which are not required is being ignored from all the APIs.....if any api is not ignoring and returning error then it should be a defect ....but I will let pawan confirm this
 like 2

> [Yesterday 1:33 PM] Rohit Kolapkar
yes, We are ignoring all extra inputs.

so if we give {"purge_data": true} or {"purge_data": false} or even  {"purge_data": abc}, all will be ignored

**Review by seagate-sarang-sawant — APPROVED** [review]

File: libs/csm/rest/csm_rest_iamuser.py (hunk 1 above)
Comment (RajeshPratapSInghChouhan) [reply libs/csm/rest/csm_rest_iamuser.py:561]: ok

**Comment by RajeshPratapSInghChouhan** [comment]
@apoorva-rao > @RajeshPratapSInghChouhan please verify CI/CD sanity and regression runs are all successful
executed : https://jts.seagate.com/browse/TEST-43083

**Pull request merged**